src/CMakeLists.txt

```diff
@@ -536,6 +536,16 @@ zeek_target_link_libraries(zeek_objs)
 
 if (HAVE_SPICY)
     target_link_libraries(zeek_objs PRIVATE hilti spicy)
+
+    # On some platforms the transitive link of the Spicy and HILTI object
+    # libraries via e.g., `hilti` and `zeek_objs` into `zeek_exe` does not work,
+    # see #3177. Make it explicit to be sure.
+    if (BINARY_PACKAGING_MODE)
+        target_link_libraries(
```

Comment:
I am not really sure whether this extra conditional makes this slip through CI, but would still fail similar to the last failing run on this PR on some platforms, but at least from what I see this fixes the issue I was seeing with the reproducer and custom-built packages. @0xxon, would you mind taking this on a test run through the packaging pipeline?